Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -171,8 +171,13 @@ const std::string MatrixWorkspace::toString() const {
   std::ostringstream os;
   os << id() << "\n"
      << "Title: " << getTitle() << "\n"
-     << "Histograms: " << getNumberHistograms() << "\n"
-     << "Bins: " << blocksize() << "\n";
+     << "Histograms: " << getNumberHistograms() << "\n";
+
+  try {
+    os << "Bins: " << blocksize() << "\n";
+  } catch (std::length_error) {
+    os << "Bins: variable\n"; // TODO shouldn't use try/catch
```

Comment:
So is this try-catch still required?

Reply:
Since there isn't currently a good way to check if `blocksize()` will work versus raise an exception, it is still needed.

Reply:
Ok. Also I think this should catch should be modified to catch by reference, but I don't want to hold this up.